### PR TITLE
Remove repositories from pom

### LINKED ISSUE #11: repositories in pom
repositories should not be added to the pom
It is not advised to put the repositories in the pom
cc @joshmoore

### PR DESCRIPTION
**Pull request by jburel**
Fixes https://github.com/ome/omero-artifact-plugin/issues/11

To test locally
* Go to a repo e.g. ``omero-common``, in ``build.gradle`` set the version of ``"org.openmicroscopy.project"`` to ``"5.5.12-SNAPSHOT"``
* run ``gradle publishToMavenLocal``
* check in the ``.m2`` directory ``pom`` file for omero-common. Check that there is no repository tag in the pom file

no longer add the repository tag in the pom
cc @joshmoore

### CODE REVIEW

**Comment by sbesson** [comment]
Assuming this passes the daily builds without issue, I concur a local `.m2` repository should never be listed under `repositories` in a `pom.xml`

**Comment by jburel** [comment]
I don't think the daily build update the "infrastructure" package
e.g. 
```
plugins {
    id "java-library"
    id "org.openmicroscopy.project" version "5.5.1"
    id "org.openmicroscopy.dsl" version "5.5.0"
}
```
in ``build.gradle`` of ``omero.model`` will not be updated
so this PR will need to be evaluated locally

File: src/main/kotlin/org/openmicroscopy/PublishingPlugin.kt [hunk 1]
```diff
@@ -65,19 +65,7 @@ class PublishingPlugin : Plugin<Project> {
         apply<MavenPublishPlugin>()
         apply<ArtifactoryPlugin>()
     }
-
-    // ORIGINAL
-    /* jar {
-        manifest {
-            attributes(
-                    'Build-Timestamp': new java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(new Date()),
```
Comment (joshmoore) [inline src/main/kotlin/org/openmicroscopy/PublishingPlugin.kt:73]: :+1: for removing commented code. For reference, these are the current values (i.e. they're being set _somewhere_):
```
Manifest-Version: 1.0
Implementation-Title: jar
Implementation-Version: 5.5.6-SNAPSHOT
Built-By: jamoore
Built-Date: 26/03/2020
Class-Path: ...
Built-Gradle: 5.2.1
Built-JDK: 1.8.0_202-ea
```

**Review by joshmoore — APPROVED** [review]
Before and after builds:

```
/opt/omero-build/omero-model $ cat ~/.m2/repository/org/openmicroscopy/omero-model/5.5.6-SNAPSHOT/omero-model-5.5.6-SNAPSHOT.pom  | grep jamoore
      <url>file:/Users/jamoore/.m2/repository/</url>
/opt/omero-build/omero-model $ vi build.gradle
/opt/omero-build/omero-model $ git diff
diff --git a/build.gradle b/build.gradle
index cc589ad1..7a57d5eb 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,6 +1,6 @@
 plugins {
     id "java-library"
-    id "org.openmicroscopy.project" version "5.5.1"
+    id "org.openmicroscopy.project" version "5.5.2-SNAPSHOT"
     id "org.openmicroscopy.dsl" version "5.5.0"
 }
/opt/omero-build/omero-model $gradle clean publishToMavenLocal
...
BUILD SUCCESSFUL in 10s
17 actionable tasks: 17 executed
/opt/omero-build/omero-model $ cat ~/.m2/repository/org/openmicroscopy/omero-model/5.5.6-SNAPSHOT/omero-model-5.5.6-SNAPSHOT.pom  | grep jamoore
/opt/omero-build/omero-model $
```

File: src/main/kotlin/org/openmicroscopy/PublishingPlugin.kt (hunk 1 above)
Comment (jburel) [reply src/main/kotlin/org/openmicroscopy/PublishingPlugin.kt:73]: Values are being set, the commented section was more for reference when the plugin was written to be sure that the values were captured

Comment (jburel) [reply src/main/kotlin/org/openmicroscopy/PublishingPlugin.kt:73]: values are set with that code
```
    private
    fun Project.configureManifest() {
        plugins.withType<JavaPlugin> {
            tasks.named<Jar>("jar") {
                manifest {
                    attributes["Implementation-Title"] = name.replace(Regex("[^A-Za-z0-9]"), "")
                    attributes["Implementation-Version"] = project.version
                    attributes["Built-By"] = System.getProperty("user.name")
                    attributes["Built-Date"] = SimpleDateFormat("dd/MM/yyyy").format(Date())
                    attributes["Built-JDK"] = System.getProperty("java.version")
                    attributes["Built-Gradle"] = gradle.gradleVersion
                    attributes["Class-Path"] = getRuntimeClasspathConfiguration(project)
                            ?.joinToString(separator = " ") { it.name }
                }
            }
        }
    }
``

**Comment by sbesson** [comment]
Is the next step here to publish a 5.5.2 version of this Gradle plugin and test its consumption by omero-build?

**Comment by joshmoore** [comment]
WFM

**Comment by jburel** [comment]
i was going to go over https://github.com/ome/omero-artifact-plugin/pull/10 and see if there were fixed bugs that could be extracted from the changes.

**Pull request merged**